test/broadcast.jl

```diff
@@ -950,3 +950,41 @@ p0 = copy(p)
 @test map(.+, [[1,2], [3,4]], [5, 6]) == [[6,7], [9,10]]
 @test repr(.!) == "Base.Broadcast.BroadcastFunction(!)"
 @test eval(:(.+)) == Base.BroadcastFunction(+)
+
+@testset "Issue #28382: inferrability of broadcast with Union eltype" begin
+    @test isequal([1, 2] .+ [3.0, missing], [4.0, missing])
+    @test_broken Core.Compiler.return_type(broadcast, Tuple{typeof(+), Vector{Int},
+                                                            Vector{Union{Float64, Missing}}}) ==
+        Vector{<:Union{Float64, Missing}}
+    @test Core.Compiler.return_type(broadcast, Tuple{typeof(+), Vector{Int},
+                                                     Vector{Union{Float64, Missing}}}) ==
+        AbstractVector{<:Union{Float64, Missing}}
+    @test isequal([1, 2] + [3.0, missing], [4.0, missing])
+    @test_broken Core.Compiler.return_type(+, Tuple{Vector{Int},
+                                                    Vector{Union{Float64, Missing}}}) ==
+        Vector{<:Union{Float64, Missing}}
+    @test Core.Compiler.return_type(+, Tuple{Vector{Int},
+                                             Vector{Union{Float64, Missing}}}) ==
+        AbstractVector{<:Union{Float64, Missing}}
+    @test_broken Core.Compiler.return_type(+, Tuple{Vector{Int},
+                                                    Vector{Union{Float64, Missing}}}) ==
+        Vector{<:Union{Float64, Missing}}
+    @test isequal(tuple.([1, 2], [3.0, missing]), [(1, 3.0), (2, missing)])
+    @test_broken Core.Compiler.return_type(broadcast, Tuple{typeof(tuple), Vector{Int},
+                                                            Vector{Union{Float64, Missing}}}) ==
+        Vector{<:Tuple{Int, Any}}
+    @test Core.Compiler.return_type(broadcast, Tuple{typeof(tuple), Vector{Int},
+                                                     Vector{Union{Float64, Missing}}}) ==
+        AbstractVector{<:Tuple{Int, Any}}
```

Comment:
#39618 seems to fix these tests. @nalimilan Could that mean that this workaround might not even be needed anymore? Or does it maybe just fix this particular example?

Reply:
Ah then it's great! Actually all the work I did in this PR had not effect for now due to this inference failure (a regression due to changes to reduce compile time introduced since 1.5). So if you can replace these `@test_broken` with `@test` then it's perfect!

Reply:
Ah, cool. Thanks for checking!